active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,220 @@
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+### Add formatting
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP addresss" 
+    Header="Enter equation:" 
+    PlaceholderText="000.000.000.000"
+    FormatType="IPAddress" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for international telephone" 
+    Header="Enter equation:" 
+    PlaceholderText="0 (123) 456-7890"
+    FormatType="InternationalTelephone" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for custom input" 
+    Header="Enter equation:" 
+    PlaceholderText="00 ft 00 in"
+    CustomFormatString="## ft ## in"
+    StripLeadingZeros="False" />
+```
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    Text="2"
+    StepSize="2"
+    UpDownArrowsEnabled="True"
+    UpDownDragEnabled="True" />
+```
+
+### Add a prefix and suffix 
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP address" 
+    Header="Enter equation:" 
+    PlaceholderText="000.000.000.000" 
+    Prefix="IP Address:" />
+```
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for cents" 
+    Header="Enter equation:" 
+    PlaceholderText="0.00" 
+    Suffix="cents" />
+```
+
+### Add mininum and maximum values
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    PlaceholderText="1" 
+    MinValue="1"
+    MaxValue="99" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+enum NumberBoxFormatType
+{
+    IPAddress,
+    InternationalTelephone,
+    Currency,
+};
+
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    String Prefix;
+    String Suffix;
+    
+    Boolean AcceptsCalculation;
```

Comment:
How will calculation be triggered? Automatically, on pressing the enter key, on lost focus, or some other way?
Will the change after calculation be part of the undo stack within that control, so it's possible to go back and view the original sum after the result has been displayed?

Reply:
@sonnemaf's initial gif shows calculation on loss of focus. I believe this is the most direct and intuitive form of interaction but I will need to engage with a designer and member of the Fluent Design System rep to validate this. Is there an alternative you prefer, @mrlacey?

Reply:
@SavoySchuler  Could loss of focus AND pressing Enter, trigger the calculation?

Reply:
you mean OR?

Reply:
> you mean OR?

Indeed I did mean OR

Reply:
If calculation is done on lost focus there's no way for automation/accessibility tools to report the final value without then returning to the control. This seems like an undesirable experience. 
Once focus is lost I'm also assuming that the ability to return to the original sum will also be lost from the undo stack. Maintaining the original sum in the undo stack (or not) needs to be documented in this spec. It would be nice to have if calculation is done before lost focus but if the calculation is done on lost focus then getting the original sum back is essential.

If the calculation is only done on lost focus, doesn't this also create issues for validation. For example, if calculation is done after focus is lost and it results in a value that is outside of the accepted range, what gets shown in the control? An error state? The max/min value (based on which boundary was exceeded)? - More thought is needed in this area.

Reply:
Agreed with @mdtauk that Enter should trigger the calculation. That's a commonly used shortcut in the Calculator app and I believe will be intuitive for and expected by many users. Being forced to move focus to see the results of the calculation feels cumbersome and slow.

Reply:
In OneNote it's triggered by "= " (equals then space), enter, and tab. And the calculation is its own frame in the Undo stack.

Reply:
In Adobe XD there's not need for =, you can just write simple arithmetic directly. And it's the "formula" doesn't get saved.

Reply:
Loss of focus, "=", "Enter" and stepping (hyper drag, hyper scroll, up/down arrow keys, StepButton) are currently listed as the events/key presses that trigger evaluation. Does this seem comprehensive and acceptable to everyone?

The change after calculation is currently planned to be part of the undo stack.